azure-pipelines.yml

```diff
@@ -0,0 +1,117 @@
+trigger:
+- v2
+
+jobs:
+
+  - job: FunctionsV1Tests
+    pool: 
+      vmImage: 'windows-latest'
+
+    variables:
+      solution: 'WebJobs.Extensions.DurableTask.sln'
+      buildPlatform: 'Any CPU'
+      buildConfiguration: 'Debug'
+
+    steps:
+    - task: NuGetToolInstaller@0
+
+    - task: DotNetCoreCLI@2
+      inputs:
+        command: 'restore'
+        projects: '**/**/*.csproj'
+        feedsToUse: 'config'
+        nugetConfigPath: '.nuget/nuget.config'
+
+    - task: VSBuild@1
+      inputs:
+        solution: '$(solution)'
+        platform: '$(buildPlatform)'
+        configuration: '$(buildConfiguration)'
+
+    - task: VSTest@2
+      inputs:
+        testSelector: 'testAssemblies'
+        testAssemblyVer2: |
+          **/*tests.v1.dll
+        testFiltercriteria: 'Category=FunctionsV1_BVT'
+        diagnosticsEnabled: True
+        rerunFailedTests: true
+        rerunFailedThreshold: 10
+        rerunMaxAttempts: 2
+      env:
+        AzureWebJobsStorage: $(AzureWebJobsStorage)
+
+  - job: FunctionsV2Tests
+    pool: 
+      vmImage: 'windows-latest'
+
+    strategy:
+      parallel: 9
```

Comment:
It might be the root cause why this pipeline takes around 30 min.  In this case, FunctionsV2Test run with 9 agents parallel. However, FunctionV1Test run on single agent. You don't specify this for FunctionV1Test.  I don't know how much your organization enables Azure Pipeline run parallel.  FunctionV1Test and FunctionV2Test might be the same number of the test cases. If you have only 10 agents, then you can run FunctionV1Test with 5 agents, FunctionV2Test with 5 agents as well.

Reply:
I actually intentionally left FunctionsV1 Tests on a single agent. We're only running a subset of FunctionsV1 Tests and while testing, FunctionsV1 Tests finished before any of the 9 agents associated with FunctionsV2 Tests, so it isn't actually holding back performance.